qt/scientific_interfaces/Muon/IALCDataLoadingModel.h

```diff
@@ -0,0 +1,47 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#pragma once
+
+#include "DllConfig.h"
+#include "IALCDataLoadingView.h"
+
+namespace MantidQt::CustomInterfaces {
+
+class MANTIDQT_MUONINTERFACE_DLL IALCDataLoadingModel {
+
+public:
+  virtual ~IALCDataLoadingModel() = default;
+  virtual void load(const IALCDataLoadingView *view) = 0;
+  virtual void cancelLoading() const = 0;
+  virtual Mantid::API::MatrixWorkspace_sptr exportWorkspace() = 0;
+  virtual bool checkCustomGrouping(const std::string &detGroupingType, const std::string &forwardGrouping,
+                                   const std::string &backwardGrouping) = 0;
+  virtual void updateAutoLoadCancelled() = 0;
+  virtual bool loadFilesFromWatchingDirectory(const std::string &firstFile, const std::vector<std::string> &files,
+                                              const std::string &runsText) = 0;
+  virtual std::string getPathFromFiles(std::vector<std::string> files) = 0;
+
+  // Getters
+  virtual bool getLoadingData() = 0;
+  virtual Mantid::API::MatrixWorkspace_sptr getLoadedData() = 0;
+  virtual std::vector<std::string> &getLogs() = 0;
+  virtual std::vector<std::string> &getPeriods() = 0;
+  virtual Mantid::API::MatrixWorkspace_sptr getWsForMuonInfo() = 0;
+  virtual double getMinTime() const = 0;
+  virtual std::string &getRunsText() = 0;
```

Comment:
Can any of these getters be `const`?

Reply:
Well spotted, all of them can be const, except for the ones returning by reference.